config/management/operational/src/governance.rs

```diff
@@ -80,21 +164,27 @@ impl RemoveValidator {
             .validator_config(self.input.account_address)?
             .human_name;
 
-        // Prepare a transaction to remove them from the ValidatorSet
         let seq_num = client.sequence_number(libra_root_address())?;
-        let txn = build_raw_transaction(
-            config.chain_id,
-            libra_root_address(),
+        let script = transaction_builder::encode_remove_validator_and_reconfigure_script(
             seq_num,
-            transaction_builder::encode_remove_validator_and_reconfigure_script(
-                seq_num,
-                name,
-                self.input.account_address,
-            ),
+            name,
+            self.input.account_address,
         );
 
-        let mut storage = config.validator_backend();
-        let signed_txn = storage.sign(LIBRA_ROOT_KEY, "remove-validator", txn)?;
-        client.submit_transaction(signed_txn)
+        build_and_submit_libra_root_transaction(&config, seq_num, script)
     }
 }
+
+fn build_and_submit_libra_root_transaction(
+    config: &Config,
+    seq_num: u64,
+    script: Script,
+) -> Result<TransactionContext, Error> {
+    let txn = build_raw_transaction(config.chain_id, libra_root_address(), seq_num, script);
+
+    let mut storage = config.validator_backend();
+    let signed_txn = storage.sign(LIBRA_ROOT_KEY, "remove-validator", txn)?;
```

Comment:
nit: the script name "remove-validator" needs to be fixed.. Perhaps pass it directly into this function from the caller?

Reply:
oh yeah, I saw this and forgot to comeback .... problem of working too late at night...